library/src/main/java/org/apache/apex/malhar/lib/utils/serde/SerdeCollectionSlice.java

```diff
@@ -37,35 +37,40 @@
  * @since 3.5.0
  */
 @InterfaceStability.Evolving
-public class SerdeListSlice<T> implements Serde<List<T>, Slice>
+public class SerdeCollectionSlice<T, CollectionT extends Collection<T>> implements Serde<CollectionT, Slice>
 {
   @NotNull
   private Serde<T, Slice> serde;
 
-  private SerdeListSlice()
+  @NotNull
+  private Class<? extends CollectionT> collectionClass;
+
+  private SerdeCollectionSlice()
   {
     // for Kryo
   }
 
   /**
-   * Creates a {@link SerdeListSlice}.
+   * Creates a {@link SerdeCollectionSlice}.
    * @param serde The {@link Serde} that is used to serialize and deserialize each element of a list.
    */
-  public SerdeListSlice(@NotNull Serde<T, Slice> serde)
+  public SerdeCollectionSlice(@NotNull Serde<T, Slice> serde, @NotNull Class<? extends CollectionT> collectionClass)
   {
     this.serde = Preconditions.checkNotNull(serde);
+    this.collectionClass = Preconditions.checkNotNull(collectionClass);
   }
 
   @Override
-  public Slice serialize(List<T> objects)
+  public Slice serialize(CollectionT objects)
   {
     Slice[] slices = new Slice[objects.size()];
```

Comment:
Probably not in this PR. But I'm thinking about some special class like ArraySlice which holds buffer and array of offsets in one object.  But I don't know how it will fit into this method.